File: typescript/inputDevices.ts
export default abstract class InputDevice{
    abstract input(): any;
}

export class Keyboard extends InputDevice{
    input(){
        console.log("Inputting data from keyboard...");
    }
}

export class Mouse extends InputDevice{
    input(){
        console.log("Inputting data using Mouse...");
    }
}

File: python/input_devices.py
from abc import ABC, abstractmethod

class InputDevice(ABC):
    @abstractmethod
    def input(self):
        pass

class Keyboard(InputDevice):
    def input(self):
        print("Inputting data from keyboard...")

class Mouse(InputDevice):
    def input(self):
        print("Inputting data using Mouse...")
